setup.sh

```
@@ -1,172 +1,581 @@
+readonly SCRIPT_NAME
+SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+readonly SCRIPT_DIR
+LOCK_FILE="/tmp/${SCRIPT_NAME}.lock" # Lockfile to prevent concurrent runs
+readonly LOCK_FILE
+DEFAULT_SCRIPT_LOG_FILE="/tmp/${SCRIPT_NAME}.$(date +%Y%m%d).log"
+readonly DEFAULT_SCRIPT_LOG_FILE
 
+# --- Configuration Constants ---
+# These constants are intended for use by sourced library scripts (e.g., backup.sh, git_ops.sh).
+# Export them to make them available to those scripts.
+export readonly BACKUP_DIR_BASE="${HOME}/config_backups_crimson_cascade"
```

Shellcheck (0.10.0)

[error] This applies export to the variable named readonly, which is probably not what you want. Use a separate command or the appropriate `declare` options instead.

(SC2316)